python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +725,32 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_for_ext = config_file.get("check_dirs_for_exts", {})
+    # Fix the paths (OS-dependent)
+    for path, exts in dirs_for_ext.items():
+        fixed_path = os.path.join(*path.split('/'))
+        config['check_dirs_for_exts'][fixed_path] = exts
+
     # Override default configs
     user_configs = config_file.get("configs", [])
     for pref in user_configs:
         if pref in config:
             config[pref] = user_configs[pref]
 
 
+def check_directory_files(directories_for_files, only_changed_files):
+    for directory, file_extensions in directories_for_files.items():
+        files = get_file_list(directory, only_changed_files, config["ignore"]["directories"])
+        for filename in files:
+            if not any(filename.endswith(ext) for ext in file_extensions):
+                error_details = {
+                    'wrong': filename.split('/')[-1],
+                    'dir': directory
+                }
+                yield (filename, 1, "found file with unexpected extension: {wrong} in {dir}".format(**error_details))
```

Comment:
This could be more generic like,

``` python
"... unexpected extension found for {name}. We only expect {ext} in {dir_name}".format(name=os.path.basename(filename), ext=', '.join(file_extensions), dir_name=os.path.dirname(filename)"
```

Maybe put those stuff in that dictionary before formatting the error?